### PR TITLE
Add %U and %w parsing support

### PR DESCRIPTION
**Pull request by jenniferpeshansky**
I've opened a PR to add support for non-ISO day of week and week of year. It would be nice to have the same options for parsing as we have for output, and vice versa. I've opened a separate issue to add %b and %B output support; I thought about adding that to this PR, but I figured that one warranted some more design decisions and should be separate.

### CODE REVIEW

File: parse_date.re [hunk 1]
```diff
@@ -2295,7 +2297,7 @@ timelib_time *timelib_parse_from_format_with_map(const char *format, const char
 						add_pbf_error(s, TIMELIB_ERR_NO_TWO_DIGIT_YEAR, "A two digit year could not be found", string, begin);
 						break;
 					}
-					
+
```
Comment (derickr) [inline parse_date.re:2300]: Please don't mix functional changes with white space changes in the same commit. Feel free to separate them out to a different commit in the same PR.

File: parse_date.re [hunk 2]
```diff
@@ -2159,6 +2159,8 @@ timelib_time *timelib_parse_from_format_with_map(const char *format, const char
 	int          iso_year = TIMELIB_UNSET;
 	int          iso_week_of_year = TIMELIB_UNSET;
 	int          iso_day_of_week = TIMELIB_UNSET;
+	int          week_of_year = TIMELIB_UNSET;
+	int          day_of_week = TIMELIB_UNSET;
```
Comment (derickr) [inline parse_date.re:2163]: These are defined, and used while parsing. But nothing is done with these two variables after they are assigned.

**Review by derickr — CHANGES_REQUESTED** [review]
There are a few things that are missing in this PR, most notably the functionality.

Right now, timelib only understands ISO year/weeks, and not "natural" ones, while parsing from a format. That is mostly because different people expect this to work in a different way. Sometimes it is different among cultures. This is different than the ISO weekday calendar, which is well specified.

For example the ranges that you used for parsing the `day_of_week` in this PR are `0-6`, where MongoDB's [documentation](https://www.mongodb.com/docs/manual/reference/operator/aggregation/dateToString/#std-label-format-specifiers) uses the range `1-7` for days. There is a similar issue with the week range, which your documentation says goes to 53, but the parsing rule only allows up to 52.

The `day_of_week` and `week_of_year` variables that you set, are not used to assign actual values to the `s->time` struct. For ISO year/week, this is done further down in this file, in a section with code starting with `/* Check for mixing of ISO dates with natural dates. */` and `/* Convert ISO values */`. A similar function to `timelib_date_from_isodate` for natural week/days needs to be implemented to make use of the parsed values, and to make sure they are not used together with the ISO weekday calendar or `year, month, day` variants.

The `%U` and `%w` specifiers are also specific to MongoDB — they don't really exist in `timelib` as such, as I had to dig a little into MongoDB's documentation as to what you meant with them. Being more precise would have been helpful.

Furthermore, there are no tests provided. These go into `tests/c/parse_date_from_format_test.cpp` where the `%b` and `%B` (also MongoDB specific) format specifiers are tested in tests such as `TEST(parse_date_from_format, isoDateWithoutPrefix)`.
The current tests in `timelib` defines `U` as `TIMELIB_FORMAT_EPOCH_SECONDS`, and does not assign `w`.

But the biggest issue here is come up with a right algorithm to define what the "natural" year and week actually mean.

**Comment by jenniferpeshansky** [comment]
Thank you for answering, and I apologize for opening a hasty and incomplete PR rather than just opening an issue. 
We've discussed this and decided that, as there have been no user requests for this functionality, we should leave it alone until such time as there is. 
Thank you for your patience and help.

**Pull request closed without merging**